pkg/distributors/test/DistributionScheduler.test.ts

```diff
@@ -0,0 +1,209 @@
+import { ethers } from 'hardhat';
+import { Contract, BigNumber, BigNumberish } from 'ethers';
+import { expect } from 'chai';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+import Token from '@balancer-labs/v2-helpers/src/models/tokens/Token';
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+
+import { fp } from '@balancer-labs/v2-helpers/src/numbers';
+
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import { expectBalanceChange } from '@balancer-labs/v2-helpers/src/test/tokenBalance';
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+import { advanceTime, currentTimestamp } from '@balancer-labs/v2-helpers/src/time';
+import { ZERO_BYTES32 } from '@balancer-labs/v2-helpers/src/constants';
+import { MultiDistributor } from './helpers/MultiDistributor';
+import TypesConverter from '@balancer-labs/v2-helpers/src/models/types/TypesConverter';
+import { solidityKeccak256 } from 'ethers/lib/utils';
+import { Account } from '@balancer-labs/v2-helpers/src/models/types/types';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+
+describe('Distribution Scheduler', () => {
+  let vault: Vault;
+  let distributor: MultiDistributor;
+  let scheduler: Contract;
+
+  let stakingToken: Token, stakingTokens: TokenList;
+  let distributionToken: Token, distributionTokens: TokenList;
+
+  let distributionOwner: SignerWithAddress, poker: SignerWithAddress;
+
+  before('setup signers', async () => {
+    [, distributionOwner, poker] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('deploy distributor', async () => {
+    vault = await Vault.create();
+    distributor = await MultiDistributor.create(vault);
+    scheduler = await deploy('DistributionScheduler', { args: [distributor.address] });
+  });
+
+  sharedBeforeEach('deploy tokens', async () => {
+    stakingTokens = await TokenList.create(1);
+    stakingToken = stakingTokens.first;
+
+    distributionTokens = await TokenList.create(1);
+    distributionToken = distributionTokens.first;
+
+    await distributionTokens.mint({ to: distributionOwner });
+    await distributionTokens.approve({ to: scheduler, from: distributionOwner });
+  });
+
+  const calcScheduleId = (stakingToken: Token, distributionToken: Token, owner: Account, time: BigNumberish) => {
+    const addresses = TypesConverter.toAddresses([stakingToken, distributionToken, owner]);
+    return solidityKeccak256(['address', 'address', 'address', 'uint256'], [...addresses, time]);
+  };
+
+  describe('scheduleDistribution', () => {
+    const amount = fp(1);
+
+    it('creates a scheduled distribution', async () => {
```

Comment:
Consider adding a .skip test where scheduling fails if the sender is not the owner.

Reply:
Will handle later as mentioned above